tsdb/chunks/chunk_write_queue.go

```diff
@@ -175,10 +193,16 @@ func (c *chunkWriteQueue) queueIsFull() bool {
 }
 
 func (c *chunkWriteQueue) queueSize() int {
-	c.chunkRefMapMtx.Lock()
-	defer c.chunkRefMapMtx.Unlock()
+	var size int
+
+	for shard := 0; shard < chunkRefMapShards; shard++ {
+		c.chunkRefMapMtx[shard].Lock()
+		defer c.chunkRefMapMtx[shard].Unlock()
```

Comment:
This will lock all shards sequentially, and only unlock when the outer method `queueSize` returns — is that intentional?